### PR TITLE
Add documentation for code signing custom Mattermost builds

### PR DESCRIPTION
**Pull request by yesbhautik**
Added comprehensive documentation on code signing custom Mattermost builds across multiple platforms. This update includes:

- A new file `code-signing-custom-builds.rst` with step-by-step instructions for Windows, Mac, and Linux.
- Integration of the new documentation into `other-administration-resources.rst` for easy navigation.

The documentation aims to help users reduce security warnings and ensure smoother installations by following best practices for code signing. 

#### Summary
This pull request introduces a new section in the Mattermost documentation dedicated to the code signing process for custom builds. It provides clear instructions for users to sign their builds, enhancing security and trustworthiness.

#### Ticket Link
Fixes: #3872

### CODE REVIEW

**Comment by cwarnermm** [comment]
@iyampaul - Are you open to helping identify an appropriate reviewer for this docs PR?

**Comment by cwarnermm** [comment]
@wiggin77 - Added you as a reviewer as a member of @core-reviewers. Please reassign as needed.

File: source/manage/code-signing-custom-builds.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      Open the ``.pfx`` file and import it into the Windows Certificate Store.
+
+   4. **Sign the Executable**
+
+      - Open the command prompt as Administrator.
+      - Use ``SignTool`` to sign your executable:
+
+      .. code-block:: sh
+
+         signtool sign /v /s "My" /sha1 <cert hash> /fd SHA256 /tr http://timestamp.digicert.com /td SHA256 <path-to-your-executable>
```
Comment (wiggin77) [inline source/manage/code-signing-custom-builds.rst:55]: I think we need to step to help the user determine what the cert hash value is.

Comment (yesbhautik) [reply source/manage/code-signing-custom-builds.rst:55]: amm, we can add one more step like

```
   4. **Find the Certificate Hash**

      To find the certificate hash, open a PowerShell window and run the following command:

      .. code-block:: sh

         Get-ChildItem -Path Cert:\CurrentUser\My | Select-Object -Property Thumbprint, Subject

      Locate your certificate in the list and copy the `Thumbprint` value, which is your certificate hash.
      ```

**Review by fmartingr — APPROVED** [review]
LGTM.

One small thing missing for me is example output or what should the user look for in the verification part.

**Comment by cwarnermm** [comment]
@yesbhautik - Are you open to incorporating @fmartingr's feedback on this docs PR?

File: source/guides/other-administration-resources.rst [hunk 2]
```diff
@@ -20,4 +21,5 @@ If the information above doesn’t solve your problem, look at these other resou
 * :doc:`mmctl command line tool </manage/mmctl-command-line-tool>` - Use mmctl to manage self-hosted Mattermost servers from the command line.
 * :doc:`Migration announcement email template </onboard/migration-announcement-email>` - Use this email template to notify your users about migrating to Mattermost.
 * :doc:`Advanced permissions: backend infrastructure </onboard/advanced-permissions-backend-infrastructure>` - Read our technical guide on modifying self-hosted Mattermost installations to create custom permissions schemes.
-* :doc:`Command line tools </manage/command-line-tools>` - Learn how to use the command line to manage self-hosted Mattermost servers.
\ No newline at end of file
+* :doc:`Command line tools </manage/command-line-tools>` - Learn how to use the command line to manage self-hosted Mattermost servers.
+* :doc:`Code Signing Custom Builds </manage/code-signing-custom-builds>` - Learn how to reduce security warnings from operating systems and ensures smoother, more trustworthy installations with Code Signing.
```
Comment (cwarnermm) [inline source/guides/other-administration-resources.rst:25]:
```suggestion
* :doc:`Code signing custom builds </manage/code-signing-custom-builds>` - Learn how to reduce security warnings from operating systems and ensure smoother, more trustworthy installations with Code Signing.
```

File: source/manage/code-signing-custom-builds.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
+Code Signing Custom Builds
```
Comment (cwarnermm) [inline source/manage/code-signing-custom-builds.rst:1]:
```suggestion
Code signing custom builds
```

File: source/manage/code-signing-custom-builds.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
+Code signing custom builds
+==========================
+
+.. include:: ../_static/badges/allplans-selfhosted.rst
+  :start-after: :nosearch:
+
+Code signing is an essential process for ensuring the authenticity and integrity of your custom ``Mattermost`` builds. This guide provides steps on how to code sign a build using your own certificates across different operating systems: Windows, Mac, and Linux.
```
Comment (cwarnermm) [inline source/manage/code-signing-custom-builds.rst:7]:
```suggestion
Code signing is an essential process for ensuring the authenticity and integrity of your custom ``Mattermost`` builds. This guide provides steps on how to code sign a build using your own certificates for Windows, Mac, and Linux.
```

File: source/manage/code-signing-custom-builds.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+.. important::
+
+  Code signing helps reduce security warnings from operating systems and ensures smoother, more trustworthy installations. Make sure to follow each operating system's guidelines and best practices for signing applications.
```
Comment (cwarnermm) [inline source/manage/code-signing-custom-builds.rst:11]:
```suggestion
  Make sure to follow each operating system's guidelines and best practices for signing applications.
```

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**